File: ruby/difference-of-squares/difference_of_squares.rb
class Squares
    def initialize(n)
        @n = n
    end
    def square_of_sum
        (1..@n).sum ** 2
    end
    def sum_of_squares
        (1..@n).sum{|i| i ** 2}
    end
    def difference
        self.square_of_sum - self.sum_of_squares
    end
end

module BookKeeping
    VERSION = 4 # Where the version number matches the one in the test.
end

File: fsharp/difference-of-squares/DifferenceOfSquares.fs
module DifferenceOfSquares

let squareOfSum number = [1..number] |> List.sum |> (fun x -> x * x)

let sumOfSquares number = [1..number] |> List.sumBy (fun x -> x * x)

let differenceOfSquares number = squareOfSum number - sumOfSquares number